### PR TITLE
Update GRPC packages

### LINKED ISSUE #394: Agent fails to enable infinite tracing in net5.0 docker images
**Description**
With the official .net 5.0 images, infinite tracing is broken and doesn't report data but standard tracing does.

**Expected Behavior**
After enabling infinite tracing, the tracing data is sent to the edge observer.

**[Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag](https://docs.newrelic.com/docs/using-new-relic/cross-product-functions/troubleshooting/new-relic-diagnostics) results**

```
2020-12-16 14:57:42,368 NewRelic  DEBUG: [pid: 8, tid: 18] SpanStreamingService: Error creating gRPC channel to endpoint <redacted>. (attempt 0)  - Exception: NewRelic.Agent.Core.DataTransport.GrpcWrapperException: Unable to create new gRPC Channel 
 ---> System.DllNotFoundException: Unable to load shared library 'libdl.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: liblibdl.so: cannot open shared object file: No such file or directory 
   at Grpc.Core.Internal.UnmanagedLibrary.Linux.dlopen(String filename, Int32 flags) 
   at Grpc.Core.Internal.UnmanagedLibrary.LoadLibraryPosix(Func`3 dlopenFunc, Func`1 dlerrorFunc, String libraryPath, String& errorMsg) 
   at Grpc.Core.Internal.UnmanagedLibrary..ctor(String[] libraryPathAlternatives) 
   at Grpc.Core.Internal.NativeExtension.LoadUnmanagedLibrary() 
   at Grpc.Core.Internal.NativeExtension.LoadNativeMethods() 
   at Grpc.Core.Internal.NativeExtension..ctor() 
   at Grpc.Core.Internal.NativeExtension.Get() 
   at Grpc.Core.GrpcEnvironment.GrpcNativeInit() 
   at Grpc.Core.GrpcEnvironment..ctor() 
   at Grpc.Core.GrpcEnvironment.AddRef() 
   at Grpc.Core.Channel..ctor(String target, ChannelCredentials credentials, IEnumerable`1 options) 
   at NewRelic.Agent.Core.DataTransport.GrpcWrapper`2.CreateChannel(String host, Int32 port, Boolean ssl, Metadata```
```

**Steps to Reproduce**
Setup any application with a base docker image of `mcr.microsoft.com/dotnet/aspnet:5.0.1` and try to enable infinite tracing.

**Your Environment**
Docker container based on `mcr.microsoft.com/dotnet/aspnet:5.0.1`

**Additional context**

### PR DESCRIPTION
**Pull request by amweiss**
### Description

Fixes #394 by updating GRPC nuget package to fix loading on Linux hosts for .NET 5.0.

### Testing

Ran existing tests with updated dependency.

### CODE REVIEW

**Comment by amweiss** [comment]
Ok, this is now failing at the same point as the markdown only change in #417 so I assume it's a bad test and not the library update.

File: build/build_functions.ps1 [hunk 1]
```diff
@@ -36,7 +36,7 @@ function New-HomeStructure {
         return
     }
 
-    $homeFolderPath = New-item -Path "$Path" -Name "$Name" -Type Directory -Force
```
Comment (vuqtran88) [inline build/build_functions.ps1:39]: unnecessary code changes.

File: build/build_functions.ps1 [hunk 2]
```diff
@@ -63,8 +63,8 @@ function Copy-ExtensionsInstrumentation {
         $xmlPath = $item.Value.FullName
         $xmlTargetName = $item.Key.BaseName
 
-        Copy-Item -Path "$dllPath" -Destination "$Destination\extensions" -Force
```
Comment (vuqtran88) [inline build/build_functions.ps1:66]: unnecessary code changes.

File: build/build_functions.ps1 [hunk 3]
```diff
@@ -88,7 +88,7 @@ function Copy-ExtensionsStorage {
 
     foreach ($item in $Extensions) {
         $dllPath = $item.FullName
-        Copy-Item -Path "$dllPath" -Destination "$Destination\extensions" -Force
```
Comment (vuqtran88) [inline build/build_functions.ps1:91]: unnecessary code changes.

File: build/build_functions.ps1 [hunk 4]
```diff
@@ -101,7 +101,7 @@ function Copy-ExtensionsOther {
     )
 
     $ErrorActionPreference = "Stop"
-
```
Comment (vuqtran88) [inline build/build_functions.ps1:104]: unnecessary code changes.

File: build/build_functions.ps1 [hunk 5]
```diff
@@ -112,17 +112,17 @@ function Copy-ExtensionsOther {
         return
     }
 
-    Copy-Item -Path "$RootDirectory\src\Agent\Miscellaneous\NewRelic.Providers.Wrapper.Misc.Instrumentation.xml" -Destination "$Destination\extensions" -Force
```
Comment (vuqtran88) [inline build/build_functions.ps1:115]: unnecessary code changes.

File: build/build_functions.ps1 [hunk 6]
```diff
@@ -138,34 +138,35 @@ function Copy-AgentRoot {
 
     $ErrorActionPreference = "Stop"
 
-    Copy-Item -Path "$RootDirectory\licenses\LICENSE.txt" -Destination "$Destination" -Force
```
Comment (vuqtran88) [inline build/build_functions.ps1:141]: unnecessary code changes.

File: build/build_functions.ps1 [hunk 7]
```diff
@@ -138,34 +138,35 @@ function Copy-AgentRoot {
...
+        Copy-Item -Path "$RootDirectory\src\Agent\NewRelic\Agent\Extensions\NewRelic.Agent.Extensions\bin\$Configuration\netstandard2.0\NewRelic.Agent.Extensions.dll" -Destination "$Destination" -Force
+        Copy-Item -Path "$RootDirectory\src\_build\AnyCPU-$Configuration\NewRelic.Agent.Core\netstandard2.0-ILRepacked\NewRelic.Agent.Core.dll" -Destination "$Destination" -Force
+        Copy-Item -Path "$RootDirectory\src\Agent\Miscellaneous\netcore20-agent-readme.md" -Destination "$Destination\README.md" -Force
     }
 
     $grpcDir = Get-GrpcPackagePath $RootDirectory
     if ($Linux) {
-        Copy-Item -Path "$grpcDir\runtimes\linux\native\libgrpc_csharp_ext.x64.so" -Destination "$Destination" -Force 
-        Copy-Item -Path "$RootDirectory\src\Agent\_profilerBuild\linux-release\libNewRelicProfiler.so" -Destination "$Destination" -Force 
+        Copy-Item -Path "$grpcDir\runtimes\linux-x64\native\libgrpc_csharp_ext.x64.so" -Destination "$Destination" -Force
```
Comment (vuqtran88) [inline build/build_functions.ps1:158]: 👍

File: build/build_functions.ps1 [hunk 8]
```diff
@@ -138,34 +138,35 @@ function Copy-AgentRoot {
...
+        Copy-Item -Path "$RootDirectory\src\Agent\_profilerBuild\linux-release\libNewRelicProfiler.so" -Destination "$Destination" -Force
     }
     else {
-        Copy-Item -Path "$grpcDir\runtimes\win\native\*.dll" -Destination "$Destination" -Force 
+        Copy-Item -Path "$grpcDir\runtimes\win-x86\native\*.dll" -Destination "$Destination" -Force
```
Comment (vuqtran88) [inline build/build_functions.ps1:162]: 👍

File: src/Agent/CHANGELOG.md [hunk 9]
```diff
@@ -13,12 +13,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 * **Adds [configuration](https://docs.newrelic.com/docs/agents/net-agent/configuration/net-agent-configuration) Environment Variables** <br/>
   * Adds MAX_TRANSACTION_SAMPLES_STORE - the maximum number of samples stored for Transaction Events.
-  * Adds MAX_EVENT_SAMPLES_STORED - the maximum number of samples stored for Custom Events.
```
Comment (vuqtran88) [inline src/Agent/CHANGELOG.md:16]: unnecessary code changes.unnecessary code changes.

File: src/Agent/NewRelic/Agent/Core/Core.csproj [hunk 10]
```diff
@@ -24,10 +24,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Google.Protobuf" Version="3.11.4" />
-    <PackageReference Include="Grpc" Version="2.28.1" />
-    <PackageReference Include="Grpc.Core" Version="2.28.1" />
-    <PackageReference Include="Grpc.Tools" Version="2.28.1">
+    <PackageReference Include="Google.Protobuf" Version="3.13.0" />
```
Comment (vuqtran88) [inline src/Agent/NewRelic/Agent/Core/Core.csproj:27]: 👍

**Review by vuqtran88 — CHANGES_REQUESTED** [review]
@amweiss Thanks for submitting a PR. I've done initial review on your PR. The main changes looks good. I would like to ask you to remove unnecessary changes to keep the PR small.

**Comment by amweiss** [comment]
> @amweiss Thanks for submitting a PR. I've done initial review on your PR. The main changes looks good. I would like to ask you to remove unnecessary changes to keep the PR small.

Trailing white space re-added, I would recommend you either make a single commit to fix the repo or update the editorconfig so this doesn't keep happening to people who respect the repo style settings: https://github.com/newrelic/newrelic-dotnet-agent/blob/main/.editorconfig#L11

**Comment by vuqtran88** [comment]
> > @amweiss Thanks for submitting a PR. I've done initial review on your PR. The main changes looks good. I would like to ask you to remove unnecessary changes to keep the PR small.
> 
> Trailing white space re-added, I would recommend you either make a single commit to fix the repo or update the editorconfig so this doesn't keep happening to people who respect the repo style settings: https://github.com/newrelic/newrelic-dotnet-agent/blob/main/.editorconfig#L11

Agree, but I would like to address it in a different PR for readablity purpose as the real changes was very small as comparing to other changes. I will probably do it as a follow up.

**Comment by vuqtran88** [comment]
> Ok, this is now failing at the same point as the markdown only change in #417 so I assume it's a bad test and not the library update.

I think some tests are flickers, some are not set up to be ran by external contributors yet. As far as my validation goes, your changes look great and the main problem is fixed.

**Review by vuqtran88 — APPROVED** [review]
LGTM!

**Comment by vuqtran88** [comment]
@amweiss Thanks for your contribution. Cheers !

**Pull request merged**